DependencyInjection/Configuration.php

```diff
@@ -173,9 +173,15 @@ private function configureClients(ArrayNodeDefinition $root)
                     })
                     ->thenInvalid('A http client can\'t be decorated with several of FlexibleHttpClient, HttpMethodsClient and BatchClient. Only one of the following options can be true. ("flexible_client", "http_methods_client", "batch_client")')
                 ->end()
+                ->validate()
+                    ->ifTrue(function ($config) {
+                        return $config['factory'] === 'httplug.factory.auto' && !empty($config['config']);
+                    })
+                    ->thenInvalid('You must specify a valid "factory" in order to use the "config"')
```

Comment:
IMO this message is not really clear. What does "valid config" mean in this context?